cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/resource/AwsVolumeResourceBuilder.java

```diff
@@ -320,11 +321,20 @@ public ResourceType resourceType() {
 
         DescribeVolumesRequest describeVolumesRequest = new DescribeVolumesRequest(volumes.getFirst());
         LOGGER.debug("Going to describe volume(s) with id(s): [{}]", String.join(",", describeVolumesRequest.getVolumeIds()));
-        DescribeVolumesResult result = client.describeVolumes(describeVolumesRequest);
-        ResourceStatus volumeSetStatus = getResourceStatus(result);
+        AtomicReference<ResourceStatus> volumeSetStatus = new AtomicReference<>();
+        try {
+            DescribeVolumesResult result = client.describeVolumes(describeVolumesRequest);
+            volumeSetStatus.set(getResourceStatus(result));
+        } catch (AmazonEC2Exception e) {
+            if (!"InvalidVolume.NotFound".equals(e.getErrorCode())) {
+                throw e;
+            }
+            LOGGER.debug("Volume set to deleted because: {}", e.getMessage());
```

Comment:
More likely -> 'The volume doesn't need to be deleted as it does not exist on the provider side' and I would prefer info level..

Reply:
done